src/rules/use-lifecycle-interface.ts

```diff
@@ -36,27 +45,44 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
   defaultOptions: [],
   create: (context) => {
     return {
-      [`ClassDeclaration > ClassBody > MethodDefinition > Identifier[name=/${lifecyclesRegexp}/]`](
+      [`ClassDeclaration > ClassBody > MethodDefinition > Identifier[name=/${lifecyclesPattern}/]`](
         node: TSESTree.Identifier,
       ) {
-        const clazz = findClassDeclarationNode(node)
+        const classDeclaration = findClassDeclarationNode(node)
+        const interfaces = (classDeclaration.implements ?? [])
+          .map(({ expression }) => expression)
+          .filter(isIdentifier)
+          .map(({ name }) => name)
+        const methodName = node.name as keyof typeof lifecycleMapper
+        const interfaceName = lifecycleMapper[methodName]
+
+        if (interfaces.includes(interfaceName)) return
 
-        const implementz = (clazz.implements || []).map((i) =>
-          isIdentifier(i.expression) ? i.expression.name : '',
+        const importDeclaration = findImportDeclaration(
+          classDeclaration,
+          effectsModulePath,
         )
-        const methodName = node.name
-        const interfaceName = methodName.substr(4)
-
-        if (!implementz.includes(interfaceName)) {
-          context.report({
-            node,
-            messageId,
-            data: {
-              methodName,
+        const {
```

Comment:
Can we move this to inside the fixer scope?
This way this only gets executed when someone wants to fix the the code.

Reply:
Nice catch. I'll update it once #70 gets merged.

Reply:
Done!